src/dht_proxy_client.cpp

```diff
@@ -252,6 +252,7 @@ DhtProxyClient::get(const InfoHash& key, GetCallback cb, DoneCallback donecb, Va
     o.req = req;
     o.finished = finished;
     o.thread = std::thread([=](){
+        auto stop_cb = std::make_shared<std::atomic_bool>(false);
```

Comment:
```suggestion
        struct GetState{ std::atomic_bool ok {true}; std::atomic_bool stop {false}; };
        auto state = std::make_shared<GetState>();
```

Reply:
line 266: `while (restbed::Http::is_open(req) and not *finished and not state->stop) {`
line 268: `if (*finished or state->stop)`

Reply:
Done :) Indeed we can group these two options.

Note: Our travis build is broken because of these commit: https://github.com/savoirfairelinux/opendht/commit/b7b5152348091a44eea16da61be9fbfb09203992
Too old on the travis machine.